ext/Geant4/g4jl_application.jl

```diff
@@ -103,16 +115,32 @@ function SSDGenerator(source::SolidStateDetectors.IsotopeSource; kwargs...)
 
     data = GeneratorData(;kwargs...)
     function _init(data::GeneratorData, ::Any)
-        gun = data.gun = move!(G4SingleParticleSource())
+        data.gun = move!(G4SingleParticleSource())
+
+        # Place the SingleParticleSource as point source
         data.position = G4ThreeVector(
             source.position.x * Geant4.SystemOfUnits.meter,
             source.position.y * Geant4.SystemOfUnits.meter,
             source.position.z * Geant4.SystemOfUnits.meter
         )
-        SetParticlePosition(gun, data.position)
-        data.direction = G4ThreeVector(source.direction.x, source.direction.y, source.direction.z)
-        SetParticleMomentumDirection(GetAngDist(gun), data.direction)
-        SetMonoEnergy(gun |> GetEneDist, 0.0 * Geant4.SystemOfUnits.MeV)
+        posdist = GetPosDist(data.gun)
+        SetPosDisType(posdist, "Point")
+        SetCentreCoords(posdist, data.position)
+
+        # Define the source emission
+        d::CartesianVector = normalize(source.direction)
+        a::CartesianVector = normalize(d × (abs(d.x) == 1 ? CartesianVector(0,1,0) : CartesianVector(1,0,0)))
+        b::CartesianVector = normalize(a × d)
+        data.direction = G4ThreeVector(d.x, d.y, d.z)
+
+        angdist = GetAngDist(data.gun)
+        SetAngDistType(angdist, "iso")
+        DefineAngRefAxes(angdist, "angref1", G4ThreeVector(a...))
+        DefineAngRefAxes(angdist, "angref2", G4ThreeVector(b...))
+        SetMinTheta(angdist, 0.0)
+        SetMaxTheta(angdist, ustrip(NoUnits, source.opening_angle))  # convert to radians
+
+        # SetMonoEnergy(GetEneDist(data.gun), 0.0 * Geant4.SystemOfUnits.MeV)
```

Comment:
Would the energy of isotopes be 0 if you don't run this line? That is, is `G4SingleParticleSource` by default not moving?

Reply:
I don't know and I might just have forgotten that I commented this line to see what the influence would be.
I uncommented this line